sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -4061,10 +4061,30 @@ object SessionWindowing extends Rule[LogicalPlan] {
           case s: SessionWindow => sessionAttr
         }
 
+        def isGapInvalid(child: Expression): Boolean = {
+          val calendarInterval = IntervalUtils
+            .safeStringToInterval(UTF8String.fromString(child.toString))
```

Comment:
`child.toString`? Does it work for all possible data types here?

Reply:
I think this only applies for `Literal` and we need to evaluate it, instead of calling `toString`.

Reply:
@viirya 
Thanks for reivew!

So far, I have judged two cases: static gap and `case when` expression. The case when expression filters the `BooleanType`. I'm not sure if there's anything missing.

Yes, I agree with your second point. I'm also trying to find a better way.